File: app/src/main/java/com/bkjcb/rqapplication/SearchGasUserActivity.java
package com.bkjcb.rqapplication;

import android.content.Intent;
import android.support.v4.app.Fragment;

import com.bkjcb.rqapplication.fragment.GasUserSearchFragment;
import com.bkjcb.rqapplication.interfaces.OnPageButtonClickListener;
import com.bkjcb.rqapplication.model.UserInfoResult;

import java.util.List;

/**
 * Created by DengShuai on 2020/5/12.
 * Description :
 */
public class SearchGasUserActivity extends SimpleBaseActivity {
    public static int CHOOSE_USER = 123;

    @Override
    protected int setLayoutID() {
        return R.layout.activity_gas_user_add;
    }

    @Override
    protected void initView() {
        initTopbar("选择燃气用户");
    }

    @Override
    protected void initData() {
        getSupportFragmentManager()
                .beginTransaction()
                .add(R.id.content_layout, GasUserSearchFragment.newInstance(new OnPageButtonClickListener<UserInfoResult.UserInfo>() {
                    @Override
                    public void onClick(UserInfoResult.UserInfo userInfo) {
                        Intent intent = new Intent();
                        intent.putExtra("data", userInfo);
                        setResult(RESULT_OK, intent);
                        finish();
                    }

                    @Override
                    public void onNext(List list) {

                    }
                },true))
                .commit();
    }

    public static void toActivity(Fragment fragment) {
        Intent intent = new Intent(fragment.getActivity(), SearchGasUserActivity.class);
        fragment.startActivityForResult(intent, CHOOSE_USER);
    }

    @Override
    public void onBackPressed() {
        finish();
    }
}

File: app/src/main/java/com/bkjcb/rqapplication/gasrecord/SearchGasUserActivity.kt
package com.bkjcb.rqapplication.gasrecord

import android.app.Activity
import android.content.Intent
import android.support.v4.app.Fragment
import com.bkjcb.rqapplication.base.BaseSimpleActivity
import com.bkjcb.rqapplication.R
import com.bkjcb.rqapplication.gasrecord.fragment.GasUserSearchFragment
import com.bkjcb.rqapplication.base.interfaces.OnPageButtonClickListener
import com.bkjcb.rqapplication.base.model.MediaFile
import com.bkjcb.rqapplication.treatmentdefect.model.UserInfoResult.UserInfo
import kotlinx.android.synthetic.main.with_bg_top_bar.*

/**
 * Created by DengShuai on 2020/5/12.
 * Description :
 */
class SearchGasUserActivity : BaseSimpleActivity() {
     override fun setLayoutID(): Int {
        return R.layout.activity_gas_user_add
    }

     override fun initView() {
        initTopBar("选择燃气用户",appbar)
    }

     override fun initData() {
        supportFragmentManager
                .beginTransaction()
                .add(R.id.content_layout, GasUserSearchFragment.newInstance(object : OnPageButtonClickListener<UserInfo> {
                    override fun onClick(userInfo: UserInfo) {
                        val intent = Intent()
                        intent.putExtra("data", userInfo)
                        setResult(Activity.RESULT_OK, intent)
                        finish()
                    }

                    override fun onNext(list: List<MediaFile>?) {

                    }

                }, true))
                .commit()
    }

    override fun onBackPressed() {
        finish()
    }

    companion object {
        var CHOOSE_USER = 123
        fun toActivity(fragment: Fragment) {
            val intent = Intent(fragment.activity, SearchGasUserActivity::class.java)
            fragment.startActivityForResult(intent, CHOOSE_USER)
        }
    }
}